pkg/controller/controller_instance.go

```diff
@@ -409,8 +409,11 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1beta1.ServiceIns
 		}
 	} else {
 		if toUpdate.Status.CurrentOperation != v1beta1.ServiceInstanceOperationDeprovision {
-			// Cancel any pending orphan mitigation since the resource is being deleted
-			toUpdate.Status.OrphanMitigationInProgress = false
+			// Cancel any in-progress operation or pending orphan mitigation since the resource is being deleted.
+			// Do not update the reconciled generation since the operation was aborted and not finished.
+			currentReconciledGeneration := toUpdate.Status.ReconciledGeneration
+			clearServiceInstanceCurrentOperation(toUpdate)
```

Comment:
Perhaps `clearServiceInstanceCurrentOperation` could return 'obj.Status.ReconciledGeneration` ? I am not sure about that suggestion.

Reply:
That feels a bit hacky to me; it's the previous reconciled generation that would be returned, and the only purpose would be to unset the changes from within the clear function.

I'm fine with this for now; we can make it a bit more clean with some of the refactoring work.